src/android/io/branch/BranchSDK.java

```diff
@@ -1165,7 +1130,20 @@ public void run() {
                     } else if (this.action.equals("registerView")) {
                         registerView(this.args.getInt(0), this.callbackContext);
                     } else if (this.action.equals("showShareSheet")) {
-                        showShareSheet(this.args.getInt(0), this.args.getJSONObject(1), this.args.getJSONObject(2), this.args.getString(3) != null ? this.args.getString(3) : "This stuff is awesome: ");
+                        Object userLocalization = this.args.opt(3);
+                        JSONObject localization = new JSONObject();
+                        //Limitation is that the user must override all the strings
+                        if (userLocalization != null && userLocalization instanceof JSONObject) {
+                            localization = (JSONObject) userLocalization;
+                        } else {
+                            localization.put("shareText", "This stuff is awesome:");
```

Comment:
Would you mind changing this to:
```
localization.put("shareText", this.args.getString(3) != null ? this.args.getString(3) : "This stuff is awesome: ");
```

Then I think we'll be all set to merge this guy in.